File: mobile/lib/features/course/domain/entities/user_course.dart
import 'package:equatable/equatable.dart';

import 'entities.dart';

class UserCourse extends Equatable {
  final String id;
  final String userId;
  final Course course;
  final int completedChapters;

  const UserCourse({
    required this.id,
    required this.userId,
    required this.course,
    required this.completedChapters,
  });

  @override
  List<Object?> get props => [
        id,
        userId,
        course,
        completedChapters,
      ];
}

File: new_backend/src/models/userCourse.ts

import { Document, Schema, model } from "mongoose";

export interface IUserCourse extends Document {
//   id: Schema.Types.ObjectId;
    userId : Schema.Types.ObjectId,
    courses : [Schema.Types.ObjectId]
}

const UserCourseSchema: Schema<IUserCourse> = new Schema({
    userId:{
    type: Schema.Types.ObjectId,
    ref: "User",
    required: [true,"User id is required"],
  },
  courses: {
    type: [{ type: Schema.Types.ObjectId, ref: "Course" }],
    default: []
  },

},
{
    timestamps: {
      createdAt: 'createdAt',
      updatedAt: 'updatedAt'
    }
  }
);

const UserCourse = model<IUserCourse>("UserCourse", UserCourseSchema);

export default UserCourse;
